### ISSUE TITLE
What Do You Mean I Can't Call Myself a Software Engineer? (John Speed)

### ISSUE THREAD

**Issue by wwkeyboard** [GeekdomSA/bookclub#12]
### Abstract

> In June 1998, the Texas Board of Professional Engineers
> established software engineering as a recognized engineering
> discipline and established licensing criteria specifically suited
> to software engineers. The author explains the legal issues
> involved and how they affect you.

**Comment by warmwaffles** [comment]
Wasn't lying when I said this :-(

**Comment by wwkeyboard** [comment]
It's really pertinent to the startup world where everything not design is "engineering"(i.e. https://medium.com/lets-make-things/eca8e839df62 )

**Comment by warmwaffles** [comment]
While true I think designers are getting better and that a CS degree is not required. It certainly helps to have an education in CS. However, if you want to optimize your stack and make it run faster. You need to find experience, not smarts.

**Comment by richardjortega** [comment]
Maybe I'm just the odd man out without a CS degree, but at least for web and a majority of backend work all you need is an intro to CS then a deep dive into what you need to build. But again, this could be just a programming v computer science thing...not saying my biz degree was any better ;)

**Comment by warmwaffles** [comment]
> then a deep dive into what you need to build.

Yes this works well with minimum viable product, however understanding software engineering principles and what happens internally on the computer will help you understand what you are doing and what optimizations could be made.

Best advice I ever got from a professor, "You need to stop and think about it."

**Comment by wwkeyboard** [comment]
@richardjortega This is exactly what I was talking about last week when I said your view of software development is scoped by you past work.The definition of engineering they(the TBPE) use is work that requires the application of an engineering education(I forget the exact wording, I think its in the article :wink:).  But where do you draw the line between engineering and not engineering?

@warmwaffles I'd argue that most web apps(and and increasing number of mobile apps) will never require more than the basic wiring of a DB to some HTML/JSON. Most messes I've seen are not from lack of engineering practices or understanding of the internals of their computer but from being unorganized, and like you said, hasty. "We'll put this in the tech debt queue" is a very common first step to failure.

**Comment by warmwaffles** [comment]
I really want to see a meta like article on reducing technical debt. I think we could benefit from it.

**Comment by wwkeyboard** [comment]
I agree! I'd love to read something that outlines a good approach to fixing tech debt. I know the Pragmatic Programmer has a section on not living with broken windows. But I'd love to see something that digs into how to quantify, prioritize, then justify to the project manager fixing tech debt.

**Comment by warmwaffles** [comment]
http://martinfowler.com/bliki/TechnicalDebt.html
